src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -221,6 +221,57 @@ private Map<String, FormDataBodyPart> extractFilesFromMultiPart(FormDataMultiPar
     return files;
   }
 
+  /**
+   * This endpoint updates the dataset.
+   *
+   */
+  @PUT
+  @Consumes("application/json")
+  @Produces("application/json")
+  @Path("/v3/{datasetId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON})
+  public Response updateDatasetByRegistrationSchema(
+      @Auth AuthUser authUser,
+      @PathParam("datasetId") Integer datasetId,
+      FormDataMultiPart multipart,
+      @FormDataParam("dataset") String json) {
+
+    // Validate dataset object
+    try {
+      Set<ValidationMessage> errors = jsonSchemaUtil.validateSchema_v1(json);
+      if (!errors.isEmpty()) {
+        throw new BadRequestException(
+            "Invalid schema:\n"
+                + String.join("\n", errors.stream().map(ValidationMessage::getMessage).toList()));
+      }
+
+      // get registration
+      DatasetRegistrationSchemaV1 registration = jsonSchemaUtil.deserializeDatasetRegistration(json);
```

Comment:
Hmm, so I think we shouldn't use the whole validation schema, because there are a lot of unused properties and  it allows any number of datasets even though we're just updating one. However, it might make sense to make a new schema with just the consent group, instead of the whole data submission scheme. That shouldn't be too much work, I think, and we could import that new schema in the data submission form, so consent groups are only defined in one location. Thoughts?

Reply:
I think using any schema here might be overkill. Maintaining two schemas for essentially same kind of thing is a bigger commitment to this approach than I'm comfortable with at this time. I wonder if we could instead manually parse the json into a new dataset intake record class that we can adjust specifically for this endpoint.

Reply:
I don't know if I agree with this. I don't think it would be more than what we're maintaining right now, because I think it could work like so:


DataRegistrationV1 -
```
{
   ...
   consentGroups: { type: array, items: { ref: /* consent group file */ }}
}
```
ConsentGroup - exactly like it is now.
```
{
   type: object
   properties: { ... }
}
```

This also ensures that the dataset updates don't put the dataset into an invalid state - what's to stop a user from simply updating their dataset from the backend and changing, e.g., the data submitter id, deleting key properties, etc.. 

To me, I feel like having a schema is important to making sure that the endpoint is secured and the user doesn't mess anything up. I find that a lot of our endpoints don't put a lot of bounds around user input, and schemas are super easy for doing that. They also sync really well with form validation in the frontend, which would be useful for the frontend piece of this ticket. I personally see a lot of value in them, and would argue we should move towards using them more, especially on important endpoints that are linked to a form in the app like this one.

Reply:
Designing a dataset schema is well beyond the scope of the assigned task/ticket. I'm not specifically opposed to the idea in general, but for this PR, we haven't done the homework to design and publish such a schema.

Reply:
That's totally fair, I just get some code smell from sort of just accepting whatever the user puts in as a dataset, and it would make me a lot more comfortable if there were well understood/formalized bounds around user input. And since she's already using the consentGroup from the data submission, just making that its own schema seems like a small code lift. But i totally understand if you prefer the old approach, I'm just a big advocate for schemas

Reply:
If we used a record class as the intake model, we would be controlling the format of the data which has the same effect as a schema.

Reply:
That's fair enough, I thought you were suggesting the same approach as the old endpoint. I am not opposed to the dataset intake class, so we can go for that, but I do feel like having the same schema as the create endpoint would be nice for consistency and frontend validation, but I won't hold the ticket up.